### PR TITLE
ports/stm32: Improve standby mode entry code for H743/7.

### PR DESCRIPTION
**Pull request by iabdalkader**
* Clear and mask D2 EXTIs.
* Disable all wakeup pins.
* Set correct voltage scaling level for standby mode.
* Disable debug MCU.

### CODE REVIEW

File: ports/stm32/powerctrl.c [hunk 1]
```diff
@@ -926,8 +938,17 @@ void powerctrl_enter_standby_mode(void) {
...
+    EXTI_D1->PR1 = 0x3fffffu;
+    EXTI_D1->IMR1 &= ~(0xFFFFu); // 16 lines
+
+    // Clear and mask D2 EXTIs.
+    EXTI_D2->PR1 = 0x3fffffu;
+    EXTI_D2->IMR1 &= ~(0xFFFFu); // 16 lines
+
+    // Disable all wakeup pins and clear flags.
+    PWR->WKUPEPR &= ~(0x1Fu);
+    PWR->WKUPCR |= PWR_WAKEUP_FLAG_ALL;
```
Comment (dpgeorge) [inline ports/stm32/powerctrl.c:951]: What if you enabled some wake up pins in code, to wake the MCU?  Will that be disabled?

File: ports/stm32/powerctrl.c [hunk 2]
```diff
@@ -952,6 +973,9 @@ void powerctrl_enter_standby_mode(void) {
     PWR->CSR1 |= PWR_CSR1_EIWUP;
     #endif
 
+    // Disable Debug MCU.
+    DBGMCU->CR = 0;
```
Comment (dpgeorge) [inline ports/stm32/powerctrl.c:977]: What is the reason for this change?  This affects all MCUs.

File: ports/stm32/powerctrl.c (hunk 1 above)
Comment (iabdalkader) [reply ports/stm32/powerctrl.c:951]: I'm not sure, never tested standby with wakeup pins before, it's usually used with RTC.

Comment (dpgeorge) [reply ports/stm32/powerctrl.c:951]: In that case I think it's best not to disable these wakeup pins.

File: ports/stm32/powerctrl.c (hunk 2 above)
Comment (dpgeorge) [reply ports/stm32/powerctrl.c:977]: This change seemingly goes against what you added in 32d76e5de66237eba456975095f998434413467a, where the debug module is enabled in low-power modes.

We should support debugging in low power modes if the user wants to do that.

File: ports/stm32/powerctrl.c (hunk 1 above)
Comment (iabdalkader) [reply ports/stm32/powerctrl.c:950]: Okay.

File: ports/stm32/powerctrl.c (hunk 2 above)
Comment (iabdalkader) [reply ports/stm32/powerctrl.c:977]: > This change seemingly goes against what you added in [32d76e5](https://github.com/micropython/micropython/commit/32d76e5de66237eba456975095f998434413467a), where the debug module is enabled in low-power modes.

Not really, that change is to make sure that debugging keeps working if/when the processor executes something like `WFI`, I think the debugger disconnected at that point before the processor woke up. The change here in this PR disables debug MCU on entry to standby intentionally so it doesn't draw any excess current. It doesn't really make sense to debug the MCU while it's in standby mode, it's going to reset on exit anyway, but I added a `#if defined(NDEBUG)`, so it's not executed on debug build, and you can remove this whole block it's just an extra precaution.

Comment (dpgeorge) [reply ports/stm32/powerctrl.c:977]: OK, let's leave it in with the NDEBUG guard.
